staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -220,7 +220,7 @@ func fromUnstructured(sv, dv reflect.Value) error {
 	case reflect.Interface:
 		return interfaceFromUnstructured(sv, dv)
 	default:
-		return fmt.Errorf("unrecognized type: %v", dt.Kind())
+		return fmt.Errorf("can't convert type %v -> %v", st.Kind(), dt.Kind())
```

Comment:
The source kind isn't really relevant here... the issue is that no matter the source kind, we don't handle this destination kind. This has the potential to produce errors that don't make sense like "can't convert type foo -> foo".